### PR TITLE
Update SUPPORT.md based on feedback

### PR DESCRIPTION
**Pull request by sftim**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Revise [SUPPORT.md](https://github.com/kubernetes/kubernetes/blob/ca5c522080bc62e2e6387516a605c24bb513256b/SUPPORT.md) to better serve readers.

- Mention asking for help on Slack
- Mention ~Stack Overflow~ Server Fault (which is like Stack Overflow, except that Stack Overflow is for coding questions, and a lot of Kubernetes questions are more about systems administration in some form)
- Link to [Report an issue with Kubernetes](https://github.com/kubernetes/kubernetes/issues/new/choose) (last), mainly because there's a further flow to direct people _away_ from filing GitHub issues, but also because people might genuinely be in a position to report a bug.

#### Which issue(s) this PR fixes:

Fixes # 114050

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

### CODE REVIEW

**Comment by sftim** [comment]
Inspired by https://github.com/kubernetes/kubernetes/issues/114050#issuecomment-1374419160

I didn't spot the CLA issue before, oops.

**Comment by dims** [comment]
want to formalize this @sftim (looks like it's still a draft)

**Comment by sftim** [comment]
It's a pain but I don't have a CLA for this: I cherry-picked a commit that hadn't passed CLA checks. :facepalm:

**Comment by rsoika** [comment]
I just would like to suggest again the idea to enable the Github discussion function. This makes it possible to move issues directly into the discussion. This can be very helpful for maintaining the issue list as you can move issues which are not a bug into the discussion and other users can try help.
I know this is overlapping the discussion forum at https://discuss.kubernetes.io/ but maybe this will help...?

File: SUPPORT.md [hunk 1]
```diff
@@ -4,26 +4,35 @@ Welcome to Kubernetes! We use GitHub for tracking bugs and feature requests.
...
+* [Monitoring, Logging, and Debugging](https://kubernetes.io/docs/tasks/debug/) guides
 
-* [Kubernetes on Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes)
+### Real-time Chat: For Beginners
 
-  * Here are some tips for [about how to ask good questions](https://stackoverflow.com/help/how-to-ask).
-  * Don't forget to check to see [what's on topic](https://stackoverflow.com/help/on-topic).
+* [Slack](https://kubernetes.slack.com) ([registration](https://slack.k8s.io)):
+The `#kubernetes-users` and `#kubernetes-novice` channels are common places where
+people offer support. If you are completely new to k8s, join this community.
```
Comment (NitishKumar06) [inline SUPPORT.md:16]:
```suggestion
people offer support. If you are completely new to k8s, join this community.
```
How about providing a reference for k/community repo?

File: SUPPORT.md [hunk 2]
```diff
@@ -4,26 +4,35 @@ Welcome to Kubernetes! We use GitHub for tracking bugs and feature requests.
...
-people offer support.
+* [Kubernetes on Server Fault](https://serverfault.com/questions/tagged/kubernetes)
+
+  * Here are some tips for [about how to ask good questions](https://serverfault.com/help/how-to-ask).
+  * Don't forget to check to see [what's on topic](https://serverfault.com/help/on-topic).
 
-### Forum
+### Forum: For Technical Questions
 
 * [Kubernetes Official Forum](https://discuss.kubernetes.io)
```
Comment (NitishKumar06) [inline SUPPORT.md:32]:
```suggestion
* [Kubernetes Official Forum](https://discuss.kubernetes.io)
```
What about the channel _kubernetes-users_?

**Pull request closed without merging**